web/src/components/auth/signup/AuthSignupRequestVerificationButton.tsx

```diff
@@ -0,0 +1,55 @@
+"use client";
+
+import { usePopup } from "@/components/admin/connectors/Popup";
+import { requestEmailVerification } from "@/app/auth/lib";
+import { Spinner } from "@/components/Spinner";
+import { useState } from "react";
+
+export function AuthSignupRequestVerificationButton({
+  children,
+  email,
+}: {
+  children: JSX.Element | string;
+  email: string;
+}) {
+  const { popup, setPopup } = usePopup();
+  const [isRequestingVerification, setIsRequestingVerification] = useState(false);
+
+  async function requestVerification() {
+    setIsRequestingVerification(true);
+    try {
+      const response = await requestEmailVerification(email);
+
+      if (response.ok) {
+        setPopup({
+          type: "success",
+          message: "A new verification email has been sent!",
+        });
+      } else {
+        const errorDetail = (await response.json()).detail;
+        setPopup({
+          type: "error",
+          message: `Failed to send a new verification email - ${errorDetail}`,
+        });
+      }
+    } catch (error) {
+      setPopup({
+        type: "error",
+        message: `An unexpected error occurred. Please try again later.`,
+      });
+    }
+    setIsRequestingVerification(false);
+  }
+
+  return (
+    <button
+      className="text-link"
+      onClick={requestVerification}
+      disabled={isRequestingVerification}
+    >
+      {isRequestingVerification && <Spinner />}
+      {children}
+      {popup}
```

Comment:
Are you sure `popup` should be inside `button`?

Reply:
We have a fetch functionality inside this button. We don't have an HOC component that contains this fetch to pass props to the button. They were using this popup to show an error if the fetch doesn't work. 

If we want to move this Pop. We need to make HOC with fech, and very universal common (dumb) button. To separate this functionality.